spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -274,10 +126,9 @@ public void bindInputs(ChannelBindingService channelBindingService) {
 		if (log.isDebugEnabled()) {
 			log.debug(String.format("Binding inputs for %s:%s", this.channelNamespace, this.type));
 		}
-		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.inputs.entrySet()) {
+		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.channelFactory.getInputs().entrySet()) {
```

Comment:
I don't think we need to move the channels managed by a `Bindable` move into `ChannelFactory`, but if we do, then binding should happen elsewhere. This effectively results in binding/unbinding channels multiple times if we have something like `@EnableBindings(Source.class,Sink.class)`.
My suggestion is to keep the channels of a given proxy instance inside the proxy (as it is more intuitive for something like `source.output()` ) and basically revert the move.